login-workflow/src/new-architecture/screens/VerifyCodeScreen/VerifyCodeScreenBase.tsx

```diff
@@ -0,0 +1,123 @@
+ * @param onResend function to call when the user clicks the 'resend code' button
+ * @param resendInstructions text to display ahead of the resend link/button
+ * @param resendLabel label for the resend link/button
+ * @param initialValue code used to pre-populate the field
+ *
+ * @category Component
+ */
+
+export const VerifyCodeScreenBase: React.FC<React.PropsWithChildren<React.PropsWithChildren<VerifyCodeScreenProps>>> = (
+    props
+) => {
+    const {
+        codeValidator = (code: string): boolean | string => {
+            if (code?.length > 0) {
+                return true;
+            }
+            return 'You must provide a valid code';
+        },
+        onResend,
+        resendInstructions,
+        resendLabel,
+        verifyCodeInputLabel,
+        initialValue,
+        title,
+        instructions,
+        ...otherProps
+    } = props;
+
+    const actionsProps = {
+        divider: otherProps.divider,
+        canGoNext: otherProps.canGoNext,
+        canGoPrevious: otherProps.canGoPrevious,
+        showPrevious: otherProps.showPrevious,
+        showNext: otherProps.showNext,
+        previousLabel: otherProps.previousLabel,
+        nextLabel: otherProps.nextLabel,
+        onPrevious: otherProps.onPrevious,
+        onNext: otherProps.onNext,
+        currentStep: otherProps.currentStep,
+        totalSteps: otherProps.totalSteps,
+        fullWidthButton: otherProps.fullWidthButton,
+        // @TODO: should we extend the rest of the props (CardActionsProps) or should we set this up to take in each sections props separately e.g., workflowCardProps, actionsProps, etc.
+    };
+
+    const [verifyCode, setVerifyCode] = React.useState(initialValue ?? '');
+    const [shouldValidateCode, setShouldValidateCode] = React.useState(false);
+    const [isCodeValid, setIsCodeValid] = React.useState(codeValidator(initialValue) ?? false);
+    const [codeError, setCodeError] = React.useState('');
+
+    const handleVerifyCodeInputChange = useCallback(
+        (code: string) => {
+            setVerifyCode(code);
+            const validatorResponse = codeValidator(code);
+
+            if (typeof validatorResponse === 'boolean' && validatorResponse === true) {
+                setIsCodeValid(true);
+                setCodeError('');
+            } else if (typeof validatorResponse === 'boolean' && validatorResponse === false) {
+                setIsCodeValid(false);
+                setCodeError('');
+            } else if (typeof validatorResponse === 'string') {
+                setIsCodeValid(false);
+                setCodeError(validatorResponse);
+            }
+        },
+        [codeValidator, verifyCode]
+    );
+
+    return (
+        <WorkflowCard>
+            <WorkflowCardHeader title={title}></WorkflowCardHeader>
+            <WorkflowCardBody>
+                <WorkflowCardInstructions divider instructions={instructions}></WorkflowCardInstructions>
```

Comment:
As there are no children passed to WorkflowCardInstructions, we can use self closing tag:
`<WorkflowCardInstructions divider instructions={instructions} />`